braindecode/augmentation/base.py

```diff
@@ -0,0 +1,202 @@
+        """General forward pass for an augmentation transform
+
+        Parameters
+        ----------
+        X : torch.Tensor
+            EEG input example or batch.
+        y : torch.Tensor | None
+            EEG labels for the example or batch. Defaults to None.
+
+        Returns
+        -------
+        torch.Tensor
+            Transformed inputs.
+        torch.Tensor, optional
+            Transformed labels. Only returned when y is not None.
+        """
+        out_X = X.clone()
+        if y is not None:
+            out_y = y.clone()
+        else:
+            out_y = torch.zeros(X.shape[0])
+
+        # Samples a mask setting for each example whether they should stay
+        # inchanged or not
+        mask = self._get_mask(X.shape[0])
+        num_valid = mask.sum().long()
+
+        if num_valid > 0:
+            # Uses the mask to define the output
+            out_X[mask, ...], tr_y = self.operation(
+                out_X[mask, ...], out_y[mask], *self.get_params(X, y)
+            )
+            # Apply the operation defining the Transform to the whole batch
+            if type(tr_y) is tuple:
+                out_y = tuple(tmp_y[mask] for tmp_y in tr_y)
+            else:
+                out_y[mask] = tr_y
+
+        if y is not None:
+            return out_X, out_y
+        else:
+            return out_X
+
+    def _get_mask(self, batch_size=None) -> torch.Tensor:
+        """Samples whether to apply operation or not over the whole batch
+        """
+        return torch.as_tensor(
+            self.probability > self.rng.uniform(size=batch_size)
+        )
+
+    @property
+    def probability(self):
+        return self._probability
+
+    def to_dict(self):
+        """Returns a dictionary describing the transform """
+        return {
+            "operation": type(self).__name__,
+            "probability": self.probability,
+        }
+
+
+class IdentityTransform(Transform):
+    """Identity transform
+
+    Transform that does not change the input.
+    """
+    operation = staticmethod(identity)
+
+
+class Compose(Transform):
+    """Transform composition
+
+    Callable class allowing to cast a sequence of Transform objects into a
+    single one.
+
+    Parameters
+    ----------
+    transforms: list
+        Sequence of Transforms to be composed.
+    """
+
+    def __init__(self, transforms):
+        self.transforms = transforms
+        super().__init__()
+
+    def forward(self, X, y):
+        for transform in self.transforms:
+            X, y = transform(X, y)
+        return X, y
+
+    def to_dict(self):
+        """Returns a DataFrame describing the transforms making the object"""
+        structure = list()
+        for i, transform in enumerate(self.transforms):
+            transform_struct = transform.to_dict()
+            transform_struct.update({"transform_idx": i})
+            structure.append(transform_struct)
+        res = pd.DataFrame(structure)
+        return res
```

Comment:
but this one is needed or also legacy to remove?

Reply:
Hmm, I had forgotten this. It is hence indeed useful in this context to know what is contained in a `Compose` transform (although not critical).

Reply:
I'll just write better docstrings then.

Reply:
Ok, after discussion with @tomMoral , we've agreed that even though it is a "kinda cool" feature, it has not it's place in the API. So I just removed both `to_dict` methods :)